content/en/docs/How Tos/custom_record.md

```diff
@@ -8,60 +8,32 @@ description: >
 
 As you can read in the [Record and LogLine page](/docs/core-concepts/record_implementation/),
 Baker processes objects called **records**.
-
 A `Record`, in Baker, is an interface that provides an abstraction over a record of flattened data,
 where columns of fields are indexed through integers.
-
 If the Record implementations provided by Baker doesn't fit your needs, you can create your own
 version, implementing the [`Record` inteface](https://pkg.go.dev/github.com/AdRoll/baker#Record).
 
 ## How to use a custom version of the Record
 
 Once your Record version is ready, you need to use it in your code.
-
-In order to do so, you must create and fill a [`baker.Components`](/docs/how-tos/baker_components/) struct.
+To do so, you must create and fill a [`baker.Components`](/docs/how-tos/baker_components/) struct.
+The only required field to set is the `CreateRecord`, which should return a new instance of your Record 
+(see more details at [CreateRecord](/docs/how-tos/baker_components/#createrecord)).
 
 ```go
-type Components struct {
-	Validate     ValidationFunc
-	CreateRecord func() Record
-	FieldByName  func(string) (FieldIndex, bool)
-	FieldNames   []string
+comp := baker.Components{
+	CreateRecord: func() baker.Record {
+		return &MyCustomRecord{}
+	},
 
-	// ... other fields
+	// ... other configuration
 }
 ```
 
-### Validate
-
-`Validate` is the function used to validate a record. It is called for each processed record
-unless not set or when the `[general] dont_validate_fields = true` configuration is set in
-the TOML file.
-
-Regardless of the TOML configuration, the function is passed to all components that can use
-it at their will.
-
-### `CreateRecord`
-
-`CreateRecord` is the function that creates a new record. If not set, a default function is
-used that creates a `LogLine` with `,` as field separator.
-
-The function is used internally by Baker to create new records every time a new one comes from
-the input.
-
-The function is also passed to components that can use it to create new records while processing.
-
-### `FieldByName`
-
-`FieldByName` gets a field index by its name. The function is mainly used by the components
-(that receive it during setup) to retrieve the index of a field they need for filtering or
-processing, but it is also used internally by Baker when sending fields to the output
-(when at least one field is selected in the output TOML configuration).
-
-### `FieldNames`
-
-`FieldNames []string` lists the names of the fields a `Record` can have. Since it's a slice, its
-length also indicates the maximum number of fields.
+Optionally, it is possible to set other functionality through the `baker.Components`. 
+For instance, you can provide a custom [Validation](/docs/how-tos/baker_components/#validate) function 
+or a custom [FieldByName](/docs/how-tos/baker_components/#fieldbyname) function for your own Record
+implementation.
```

Comment:
That's true these are optional to compile baker, however you have to provide `FieldNames` and `FieldByName` to run it and do something useful with it.
So I'd try to split this into 2 small parts/sentences:
 - the first in which we explain that we need to instruct Baker about the schema (how to map field names to field indices in your record, and vice versa). This is true whether you do this with TOML or by code
 - a second in which we can say a word about a Record validation function, which is optional